src/firstSuggestionDecoration.ts

```diff
@@ -0,0 +1,74 @@
+import { DecorationOptions, DecorationRangeBehavior, ExtensionContext, Position, Range, TextDocument, TextEditor, window } from "vscode";
+import getHintColor from "./inlineSuggestions/hintColor";
+import { firstSuggestionDecorationAlreadyDisplayed, setFirstSuggestionFlag } from './firstSuggestionDecorationFlag';
+import { fireEvent, getState } from "./binary/requests/requests";
+import isInTheLastHour from "./utils/time.utils";
+import { getTabnineExtensionContext } from './globals/tabnineExtensionContext';
+
+// Anchor hint to the right of the line
+const firstSuggestionDecorationType = window.createTextEditorDecorationType({
+    rangeBehavior: DecorationRangeBehavior.ClosedOpen,
+});
+const maxSmallIntegerV8 = 2 ** 30; // Max number that can be stored in V8's smis (small integers)
+
+export function getFirstCompletionDecoration(
+    document: TextDocument,
+    position: Position,
+  ): DecorationOptions[] {
+    const decorations: DecorationOptions[] = [];
+    decorations.push({
+      renderOptions: {
+        after: {
+          color: getHintColor(),
+          contentText: "👈 Press Tab to accept Tabnine's suggestion",
+          margin: "20px",
+          border: '0.5px solid',
+        },
+      },
+      range: document.validateRange(new Range(position.line, maxSmallIntegerV8, position.line, maxSmallIntegerV8)),
```

Comment:
why do you need to validate? i think its very unusual to see a file with 2^30 lines...

Reply:
What `validateRange`does is "Ensure a range is completely contained in this document". This trick allows me to calculate the range that'll allow me to anchor the banner to the end of the line. I took this from [Gitlens](https://github.com/gitkraken/vscode-gitlens/blob/e916ecb0d14bed5baa9092d37c1a8569b6a1d276/src/annotations/lineAnnotationController.ts#L313). It was necessary to fix a bug I had otherwise - sometimes the banner location and typing into the editor would somehow race and cause the banner to be rendered in the middle of the line with the suggestion on the right. This solved it.

Reply:
legit, well done for catching it 🔥